product_packaging_level_salable/models/sale_order_line.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _can_be_sold_error_condition(self):
+        self.ensure_one()
+        return self.product_packaging_id and not self.product_packaging_id.sales
+
+    @api.constrains("product_packaging_id")
+    def _check_product_packaging_can_be_sold(self):
+        for line in self:
+            if line._can_be_sold_error_condition():
+                raise ValidationError(
+                    _(
```

Comment:
You can replace the call to translate a string `(_)` with `self.env._` for getting some performance improvement in some cases. See https://github.com/odoo/odoo/pull/174844.